solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/FlowReceiverContainer.java

```diff
@@ -99,14 +116,14 @@ public UUID bind() throws JCSMPException {
 	 * Closes the bound {@link FlowReceiver}.
 	 */
 	public void unbind() {
-		Lock writeLock = readWriteLock.writeLock();
 		writeLock.lock();
 		try {
 			FlowReceiverReference flowReceiverReference = flowReceiverAtomicReference.getAndSet(null);
 			if (flowReceiverReference != null) {
-				logger.info(String.format("Unbinding flow receiver container %s", id));
+				LOGGER.info("Unbinding flow receiver container {}", id);
 				flowReceiverReference.getStaleMessagesFlag().set(true);
 				flowReceiverReference.get().close();
+				Optional.ofNullable(flowReceiverReference.getTransactedSession()).ifPresent(TransactedSession::close);
```

Comment:
this. is a replacement for 
TransactesSession ts =flowReceiverReference.getTransactedSession()
if (ts!=null) ts.close()
readability could be improved?

Reply:
Fixed.